### ISSUE TITLE
Inconsistent hyphenation / word-wrapping.

### ISSUE THREAD

**Issue by nwsm** [diegomura/react-pdf#1238]
**Describe the bug**
In some scenarios, long words are not broken up and hyphenated appropriately, leading to overflow of flex containers.

**To Reproduce**
https://react-pdf.org/repl?code=3187b0760ce02e004d09e01b02984605e181bc05031804e200ee017367be3002602584003a2021bce40e40198a007bb00d257cdd90f0022b40b260516b80e45880ca01200118802d59010e011818f382112d6a8593565546089e01d40b3061d89386d6a805b6604039ad1807002b21bb25002fa0be2822392e0d8333351d181fbe800318745506968ebea1b1a9b99aa2002bb2387e044e2d68242c188830060c0005002506001f0765000f33703957b218140f508c00c002b31fb21c1412323a16020a04001d06b53c0454ca8a80c01aad323132eafae6ea36e2a1f4cd89f9e5f5ca2dcad6f6ec53cd8020103000aa8926cc083513803003d2827893678024e30b7b10a680c059c2e573b97d563b3f8633158f878321d0b85838924945a26924999a23e6b0db7dee44a449241d4e20e856000b001588b45a28144b2552e94cb6572f942b154ae574b616486602e9388d4c0b5971d6bc712cfc4fd1e3afc21bde78b64137e267fa326c3c844f42150b5752b9543d7a2bd4ee675aee84fb79b31cef27baa92ebf6eb61f498fe1b156f649a392184d51c33d0a670fc79900170b45e2c974b65f2c572b55eacd76b75fac372b0c66cb75b6df6c773b5deecf77b7dfec0f075d8f7471d3e835c7b5d3587cd16c4d870d46e34467400dc380dce0004ac8660c96662001883dc6da023b4862d180c27a6b9c100000

**Expected behavior**
Words should always be hyphenated effectively by default.

**Desktop (please complete the following information):**
 - OS: MacOS
 - Browser: Chrome
 - React-pdf version: 2.0.0

I assume the issue here is in the hyphen package, but I could be wrong and am also open to any suggestions to avoid this. It appears to have something to do with the hyphening algorithm dealing with long strings of repeated characters.

**Comment by diegomura** [comment]
Yes. This is definitely an edge case in which `hyphen` is expecting words and clearly that's not what it expects. This makes it probably return an invalid break. Wonder if this is something that happens in real world applications. Will keep it open

**Comment by GBrachetta** [comment]
I actually have a real world case. Rendering long paragraphs with a monospaced font (FiraCode) the justification is not completely reliable and there are overflows (of max 1 character).

**Comment by amaljosea** [comment]
In my case, we need to render UUID strings and it is not wrapping as expected

**Image:**
<img width="392" alt="Screenshot 2021-05-18 at 3 13 14 PM" src="https://user-images.githubusercontent.com/26934320/118629934-af3adc80-b7eb-11eb-8ad6-c17527d576fb.png">
 
**REPL:**
[Link](https://react-pdf.org/repl?code=3187b0760ce02e00408a057025803c450298c0bc300500943807cf805030c00f0022230080b6198531165d400a02180e659a004f003619b006f61622003a0046200099080beed3a72a005430a58d3c4489b34e1b9002c30f2518013801a18f27b000d67cec80460940611011103b002e0072210c1120807730d5554a0033540c250d4d4e005a1cdcbcfc82c2828e4d2a007a5d7d0cd2ca8328512313337a99592b1b7b2717774f6f5f00a0d0b0bb34f8c4949434eacc8046002600660c00160056003600762c8c000e004e792cf9a505ac9e35f5ace5b9f5f599e599cde580063792ad0abd360fca1d6f98219244d59398dad65b2399cae0f1787cfe40b05c29e0c0b0c6c954ba57e9c7992d2e9b181ed0e301382c60170d8c06e770793d5eef4cb50be556c7fdf48096a354ca0ce658219d684f4e1fd4450d823c30009d1300994db1941e0f0ec002b3718092aabb3ab35dab54aadc49301d8dccac37eb0d16043c89296963002c766b509e4766590892f237524f8cb0410890c11b7041049567327ea532af00419729d018cc563b008006e3219140903a8342038180019539f9ab060a0b26008c78983c048388a1508460b5cd0001c6c4a2424b1b0b55871540e0e3b5218de6e706248251402c8d999bc079a460f05000194ec60208db9bce60aa322a953e98012b5980502e0d00062b211af9ec782a22150e82c1958807f4d000)  (Try editing the code if it shows error 'You are not rendering a valid document')

**Comment by diegomura** [comment]
Have you tried setting a [custom hyphenation callback](https://react-pdf.org/advanced#hyphenation)? Otherwise, you can add soft-hyphens chars on the possible breaking points of your text. That will bypass the default hyphenation engine and take those as valid places. Check [here](https://react-pdf.org/repl?code=3187b0760ce02e0040ca07901880540fa0090268014305100e460178601c805af20281b44960114057012c00f10a014d498005004a5200f904d183000f001110c05805b1e60a28c95264e008601ccfb4009e006c78900dec6cc40074008c4001323017c356add250f0eb1af98585ad88405d80058f0e938f0013800d0c3d8eb000359eac480b181380308809882c401739118f09a1403bb92bab948019a70f1387a796802d17774f6f5f7f5f66a7b4803d0f9f9b70f8ff9429a070686ccdada4745c62725a46564e7e6149792c4b6d7d53470b64fb4023001300330f000b002b001b003b0745a22a262e012115c3c000700139ec1d5b938ee5f1fba1b078222b8742f578751e3757abcae8f2bbbd1e00062250cbc635f3a849526f39260014b02d6c61159446209248a5d2996c9e40a455286478aa13a359aad4a5696e0f547bc6020f04c0a177180a2de300c5627178c2713da323244dc5d4bf2d296f310a334d11167add95b2e6ed7907228e8c006614c0ce1771548743a58800ad526006a0762c1d0f868301d4834c0b154bfb63d1d8f8458f606aa754c070ac5d3467b2c51e4606bd88b0d3d23c5846361143345160350dfa8a70c46ba031b546f2450a8d41a21001b8e80c6809ae6107e1c14d70488f0a0b66011c74bc01059348e17314606bcf0001da24e362bab77767a695cf14d2ad595b9dd68aa6c27141c25bab9132f9e250e838001923cf0202dc6e0fc6057068571073a000252898028070590905b08e1c8e2011a4561386e0f811944482e82000)

**Comment by amaljosea** [comment]
@diegomura 

- We had an option to remove the hyphens from those UUIDs and use space instead. So we are displaying like `123e4567 e89b 12d3 a456 426614174000` (green) now.
- SOFT_HYPHEN looks handy, will try to use that moving forward.

Thanks 😊

**Comment by SatorCube** [comment]
Another real-world case where I am seeing this issue is with URLs. They frequently have UUID-like portions.

**Comment by nbouvrette** [comment]
We are having similar issues where we have user-generated content and random text overflowing from our PDFs rather than just having normal overflow makes our PDF look buggy.

**Comment by lexxwork** [comment]
yet another [example](https://react-pdf.org/repl?code=1887b07602e07404e0a6073025819c271802409e0070059c60086112e00c2c4036d4046c40c60358014ac0ee20c0098094000802f003e0101bc0140081f02005718600406d2ebc02e806e49017cf8ec98dc1a0169b353829844e9038e46248c06005cb664c8e487847cee01c801180158001970003c020068ec65089011f021034223a2e33c04e9b878dc0582a2cc41a87c73a8999962ed74b205d123523d3d1a200104ca10c1031889d0606bb200cdc02001949000bce1dd43eaeaede0780054e09bdca5b38da9b90396860574f50c1d4c0165b000444119e4016dfa6d5905440558ec00786eef1f2045e2024f80015880838003b2324f800d490700e198201638109c4e64b0a0a00e081385c305d242a1d0b59349128b4462ac5036a1301c4cfa92202276808ca2e01081860d7584004ce20401e89944e249379e4cb2539198d81c156bcc271004b8620c13ec2de68ac58cad58260c4021403522c0401204d5afa50ae108ed46b41e0a246a7e0f7e8020c924900094e04c08302aec039581f2305627d2eaebf9f305220310000)
of unexpected hyphenation:
  ![expect](https://i.imgur.com/qoQptdh.png)   ![real](https://i.imgur.com/CRGrmVx.png)

looks like hyphenation happens in other code place
